code/modules/vehicle/janicart.dm

```diff
@@ -44,9 +44,14 @@
 		var/turf/tile = loc
 		if(isturf(tile))
 			tile.clean_blood()
+			if(istype(tile, /turf/simulated/floor))
+				var/turf/simulated/floor/F = tile
+				F.dirt = 0
 			for(var/A in tile)
-				if(is_cleanable(A))
-					qdel(A)
+				if(istype(A, /obj/effect))
+					if(is_cleanable(A))
```

Comment:
no need to check for if it's an effect and cleanable. Just the above for the F.dirt = 0 is necessary.